tests/e2e/taqueria-plugin-ipfs-pinata.spec.ts

```diff
@@ -1,132 +1,186 @@
+		await exists('./test-project/.taq/config.json');
 
-		await configureForTests();
+		const env_file =
+			'pinataJwtToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySW5mb3JtYXRpb24iOnsiaWQiOiI1M2FkZjkxZi0yY2Q0LTQ3ZGUtYThlOS00YmM0YjI5NDI4NzYiLCJlbWFpbCI6Im1pY2hhZWxrZXJuYWdoYW5AZWNhZGxhYnMuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsInBpbl9wb2xpY3kiOnsicmVnaW9ucyI6W3siaWQiOiJOWUMxIiwiZGVzaXJlZFJlcGxpY2F0aW9uQ291bnQiOjF9XSwidmVyc2lvbiI6MX0sIm1mYV9lbmFibGVkIjpmYWxzZSwic3RhdHVzIjoiQUNUSVZFIn0sImF1dGhlbnRpY2F0aW9uVHlwZSI6InNjb3BlZEtleSIsInNjb3BlZEtleUtleSI6ImNiOGMzMzVkN2RhOThiZGQ2MTRmIiwic2NvcGVkS2V5U2VjcmV0IjoiZWIwYmUxYjRhYzNhZjE5ZGE3MjQ3MTAxNmFlZjFjNTllNjQzNTdlOTcwZmY1ZmYxZDBjNmU1ZTBkYmI1ODkzMCIsImlhdCI6MTY2OTEzMzI2MH0.U9JyTzEviH5y-GxSmD6YEkCVkcjsSX6d13a9fHE70LM';
+		await writeFile('./test-project/.env', env_file);
 
-		try {
-			await exec(`taq -p ${taqueriaProjectPath}`);
-		} catch (_) {}
+		const {} = await execute('taq', 'install ../taqueria-plugin-ipfs-pinata', './test-project');
+		await exists('./test-project/node_modules/@taqueria/plugin-ipfs-pinata/index.js');
 
-		await exec(`cp -r e2e/data/ipfs ${taqueriaProjectPath}/ipfs`);
-	});
+		const { stderr } = await execute('taq', 'publish', './test-project');
+		expect(stderr).toEqual(expect.arrayContaining(['path was not provided']));
 
-	test('ipfs pinata plugin should warn if no file specified for publish', async () => {
-		const publishRun = await exec(`taq publish`, { cwd: taqueriaProjectPath });
-		expect(publishRun.stderr).toBe('path was not provided\n');
+		await cleanup();
 	});
 
-	// Skipping this test until issue https://github.com/ecadlabs/taqueria/issues/1188 is resolved
-	test.skip('ipfs pinata plugin should warn if no filehash specified for pin', async () => {
-		const publishRun = await exec(`taq pin`, { cwd: taqueriaProjectPath });
-		expect(publishRun.stderr).toBe('ipfs hash was not provided\n');
-	});
+	test('pin will error if no filehash specified', async () => {
+		// note the error is not what is expected, but it is what is returned
+		const { execute, cleanup, exists, writeFile } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
+		await exists('./test-project/.taq/config.json');
+
+		const env_file =
+			'pinataJwtToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySW5mb3JtYXRpb24iOnsiaWQiOiI1M2FkZjkxZi0yY2Q0LTQ3ZGUtYThlOS00YmM0YjI5NDI4NzYiLCJlbWFpbCI6Im1pY2hhZWxrZXJuYWdoYW5AZWNhZGxhYnMuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsInBpbl9wb2xpY3kiOnsicmVnaW9ucyI6W3siaWQiOiJOWUMxIiwiZGVzaXJlZFJlcGxpY2F0aW9uQ291bnQiOjF9XSwidmVyc2lvbiI6MX0sIm1mYV9lbmFibGVkIjpmYWxzZSwic3RhdHVzIjoiQUNUSVZFIn0sImF1dGhlbnRpY2F0aW9uVHlwZSI6InNjb3BlZEtleSIsInNjb3BlZEtleUtleSI6ImNiOGMzMzVkN2RhOThiZGQ2MTRmIiwic2NvcGVkS2V5U2VjcmV0IjoiZWIwYmUxYjRhYzNhZjE5ZGE3MjQ3MTAxNmFlZjFjNTllNjQzNTdlOTcwZmY1ZmYxZDBjNmU1ZTBkYmI1ODkzMCIsImlhdCI6MTY2OTEzMzI2MH0.U9JyTzEviH5y-GxSmD6YEkCVkcjsSX6d13a9fHE70LM';
+		await writeFile('./test-project/.env', env_file);
```

Comment:
The implementation is married to a file instead of a stream, boo

Reply:
This is the easiest way to deal with this, sadly.